pkg/operator/ceph/cluster/nodedaemon/reconcile.go

```diff
@@ -248,7 +249,9 @@ func (r *ReconcileNode) createOrUpdateNodeDaemons(node corev1.Node, tolerations
 			logger.Debugf("crash collector successfully reconciled for node %q. operation: %q", node.Name, op)
 		}
 	}
-	if cephVersion.IsAtLeast(MinVersionForCephExporter) && !cephCluster.Spec.Monitoring.MetricsDisabled {
+	exporterSockFilePath := sockDir + "/ceph-client.ceph-exporter.asok"
+	_, err := os.Stat(exporterSockFilePath)
+	if cephVersion.IsAtLeast(MinVersionForCephExporter) && !cephCluster.Spec.Monitoring.MetricsDisabled && !os.IsNotExist(err) {
```

Comment:
!os.IsNotExist(err) -> os.IsNotExist(err) as you are already checking `isNotExit` so if it is `true`  then we should reconcile as the file is not present (as per commit message)